com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/CustomMetricsSink.java

```diff
@@ -0,0 +1,168 @@
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.InfluxDB.ConsistencyLevel;
+import org.influxdb.dto.BatchPoints;
+import org.influxdb.dto.Point;
+
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingInput;
+import com.ibm.streams.operator.Tuple;
+import com.ibm.streams.operator.model.InputPortSet;
+import com.ibm.streams.operator.model.InputPortSet.WindowMode;
+import com.ibm.streams.operator.model.InputPortSet.WindowPunctuationInputMode;
+import com.ibm.streams.operator.model.InputPorts;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.Parameter;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+
+@PrimitiveOperator(name="CustomMetricsSink", namespace="com.ibm.streamsx.metrics",
+description="Java Operator CustomMetricsSink")
+@InputPorts({@InputPortSet(description="Port that ingests tuples", cardinality=1, optional=false, windowingMode=WindowMode.NonWindowed, windowPunctuationInputMode=WindowPunctuationInputMode.Oblivious), @InputPortSet(description="Optional input ports", optional=true, windowingMode=WindowMode.NonWindowed, windowPunctuationInputMode=WindowPunctuationInputMode.Oblivious)})
+@Libraries({
+	// As described here:
+	// http://www.ibm.com/support/knowledgecenter/en/SSCRJU_4.2.0/com.ibm.streams.dev.doc/doc/jmxapi-start.html
+	// Environment variables (@...@) are evaluated during compile-time and must
+	// be identical in the run-time environment.
+	"@STREAMS_INSTALL@/lib/com.ibm.streams.management.jmxmp.jar",
+	"@STREAMS_INSTALL@/lib/com.ibm.streams.management.mx.jar",
+	"@STREAMS_INSTALL@/ext/lib/jmxremote_optional.jar",
+	"opt/downloaded/*"
+	})
+public class CustomMetricsSink extends AbstractOperator {
+	
+	// Parameter definition.
+	private static final String ATTRIBUTE_NAMES = 
+			"Specifies the attributes values to output to databases.";
+
+	String[] attributeList = null;
+	
+	@Parameter(
+			optional=false,
+			description=CustomMetricsSink.ATTRIBUTE_NAMES
+			)
+	public void set_attribute_names(String listString) {
+		attributeList = listString.split(",");
+	}
+	
+	// InfluxDB variables.
+	InfluxDB influxDB = null;
+	String dbName = null;
+	BatchPoints batchPoints = null;
+	String jobName = null;
+	
+	// ElasticSearch variables.
+	TransportClient client = null;
+	XContentBuilder builder = null;
+	IndexResponse response = null;
+	
+	@SuppressWarnings("resource")
+	@Override
+	public synchronized void initialize(OperatorContext context)
+			throws Exception {
+		super.initialize(context);
+        Logger.getLogger(this.getClass()).trace("Operator " + context.getName() + " initializing in PE: " + context.getPE().getPEId() + " in Job: " + context.getPE().getJobId() );
+	
+        // Connect to InfluxDB server.
+        influxDB = InfluxDBFactory.connect("http://localhost:8086", "admin", "admin");
```

Comment:
Streams is running on several servers. So, what if your influx DB is not running on the local host. You cannot and must not assume that the user/password is admin/admin.